pkg/kubelet/prober/prober.go

```diff
@@ -115,9 +115,21 @@ func (pb *prober) probe(ctx context.Context, probeType probeType, pod *v1.Pod, s
 		pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, events.ContainerProbeWarning, "%s probe warning: %s", probeType, output)
 		klog.V(3).InfoS("Probe succeeded with a warning", "probeType", probeType, "pod", klog.KObj(pod), "podUID", pod.UID, "containerName", container.Name, "output", output)
 	} else {
+		switch result {
+		case probe.Success:
+			pb.recordContainerEvent(pod, &container, v1.EventTypeNormal, events.ContainerHealthy, "%s probe succeeded", probeType)
+		case probe.Failure:
+			pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, events.ContainerUnhealthy, "%s probe failed: %s", probeType, output)
+		default:
+			// log the unknown status
+			klog.Warningf("Unknown probe result: %s", result)
+			// record an event to expose the unknown status
+			pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, events.ContainerProbeWarning, "Unknown %s probe status: %s", probeType, result)
+		}
```

Comment:
Looking at the code, these should be handled above by the code block above: https://github.com/kubernetes/kubernetes/blob/861be66619b6a6c3a85455e88b4a337d0cecac86/pkg/kubelet/prober/prober.go#L102-L113
So I don't think you will hit this code path as it is right now